containers/src/Data/Set/Internal.hs

```diff
@@ -1823,11 +1823,22 @@ splitRoot orig =
 -- @
 --
 -- @since 0.5.11
+
+-- Proof of complexity: step executes n times. At the ith step,
+-- "insertMin x `mapMonotonic` pxs" takes O(2^i log i) time since pxs has size
+-- 2^i - 1 and we insertMin into its elements which are sets of size <= i.
+-- "insertMin (singleton x)" and "`glue` pxs" are cheaper operations that both
+-- take O(i) time. Over n steps, we have a total cost of
+--
+--   O(\sum_{i=0}^{n-1} 2^i log i)
+-- = O(log n * \sum_{i=0}^{n-1} 2^i)
+-- = O(2^n log n)
```

Comment:
`O(\sum_{i=0}^{n-1} 2^i log i)` can't possibly be right, because it has `log 0` in it.

Reply:
Yeah, that's a little icky. Updated to count from 1, step 0 simply runs in O(1).